pkg/supportbundle/spec.go

```diff
@@ -144,7 +144,12 @@ func CreateRenderedSpec(app apptypes.AppType, sequence int64, kotsKinds *kotsuti
 	renderedSpec = b.Bytes()
 
 	secretName := GetSpecSecretName(app.GetSlug())
-	existingSecret, err := clientset.CoreV1().Secrets(util.PodNamespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+	// to add the troubleshoot labels to existing support-bundle spec secrets that were created in older versions of kots
+	existingSecret, err := k8sutil.AddLabelsToSecret(clientset, util.PodNamespace, secretName, kotstypes.GetTroubleshootLabels())
+	if err != nil {
+		errors.Wrap(err, "failed to add troubleshoot labels to existing spec secret")
+	}
+
```

Comment:
I think this was originally added in commit c08ae3d because we were querying for the existing secret by the troubleshoot label only.  We have logic just below this that adds the labels to the secrets (on create or update).  It was added in https://github.com/replicatedhq/kots/pull/3539.

I might be misunderstanding the purpose of this PR, but would this mean than we don't need _any_ of these changes?  Let me know what you think.